File: CPP/1929_find_out_prime/find_out_prime.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    int M, N;
    scanf("%d %d", &M, &N);

    int arr[1000001] = {};

    for(int idx = 2; idx <= sqrt(N) + 1; idx++) //에라토스테네스의 체알고리즘
    {
        if (arr[idx] == 0){
            int jdx = 2;
            while (idx * jdx <= N)
            {
                arr[idx * jdx] = 1;
                jdx++;
            }
        }
    }
    for (int idx = M; idx <= N; idx++)
    {
        if (arr[idx] == 0)
                cout << idx << endl;
    }
}

File: python/1929_find_out_prime/find_out_prime.py
# import time
import math

#1번
# start = time.time()
M, N = map(int, input().split())
# # arr = [True for x in range(1000001)]
# check = True
# # arr[1] = False
# for target in range(M, N + 1):
#     if target == 1:
#         continue
#     for idx in range(2, int(math.sqrt(target)) + 1):
#         if (target % idx == 0):
#             check = False
#             # arr[target] = False
#             break
#     if check:
#         print(target)
#     check = True

# # for i in range(M, N + 1):
# #     if arr[i]:
# #         print(i)

# end = time.time()

# print(f"{end - start:.5f} sec")


# 2번

arr = [True for x in range(1000001)]
arr[1] = False
for target in range(2, int(math.sqrt(N)) + 1):
    if arr[target]:
        idx = 2
        while target * idx <= N:
            arr[target * idx] = False
            idx += 1

for i in range(M, N + 1):
    if arr[i]:
        print(i)


# 방식 메모리	  시간	
# 1   40832  592	
# 2   32540  5360